crates/sui-indexer/migrations/2023-03-20-041133_epoch/up.sql

```diff
@@ -40,10 +40,26 @@ CREATE INDEX epochs_end_index ON epochs (epoch_end_timestamp ASC NULLS LAST);
 -- update epoch_network_metrics on every epoch
 CREATE OR REPLACE FUNCTION refresh_view_func() RETURNS TRIGGER AS
 $body$
+DECLARE
+    attempts INT := 0;
 BEGIN
     IF (TG_OP = 'INSERT') THEN
-        REFRESH MATERIALIZED VIEW epoch_network_metrics;
-        REFRESH MATERIALIZED VIEW epoch_move_call_metrics;
+        LOOP
+            BEGIN
+                attempts := attempts + 1;
+                REFRESH MATERIALIZED VIEW epoch_network_metrics;
+                REFRESH MATERIALIZED VIEW epoch_move_call_metrics;
+                EXIT;
+            EXCEPTION
+                WHEN OTHERS THEN
```

Comment:
thought: do you know what the specific exception is? We can retry on those and fail fast on others

Reply:
btw did you modify this file manually or is there some corresponding diesel thing? Not sure how this works